DP-39093 service card api fix default drupal translations for service card api content type

dstorozhuk opened the pull request:
**Description:**
1. Renamed /node/{node}/translations, /media/{media}/translations > /node/{node}/mass-translations, /media/{media}/mass-translations.
2. Enabled content_translation module and added fields translation for Service Cards fields.
3. Added a new language negotiator from contrib modules which meets our need: to don’t apply for outbond URLs but allow to swicth content language.


**Jira:** (Skip unless you are MA staff)
DP-****


**To Test:**
- [ ] Add steps to test this feature


---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

Commit:
new content type

dstorozhuk committed:
DP-38918: Add API Service Card bundle and content moderation workflow

Introduced the `ApiServiceCardBundle` class for managing node entities and updated the `mass_content_entity_bundle_info_alter()` function to register it. Added a content moderation workflow specific to API Service Cards and configured form displays, field groupings, and weights. Additionally, updated language settings and field properties to align with requirements.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-38918: Add view configuration for API Service Cards

Introduced a new view configuration to list and manage API Service Cards. The view includes both administrative and default displays with fields like Title, Created, and Changed, and is accessible via a dedicated admin page at "admin/content/api-service-cards".

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

Commit:
view enhancements

dstorozhuk committed:
DP-38918: Update API Service Cards view and add MMG Editor role

Renamed menu and updated paths in the API Service Cards view to align with the mass dashboard. Introduced a new MMG Editor role with specific permissions and actions for assigning/removing the role from users. Adjusted language type configurations for consistency.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Enable content_translation.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Enable content translation for API Service Card nodes

This update adds content_translation module dependencies and settings to the configuration. It ensures API Service Card nodes support multilingual content.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Refine entity validation in mass_translations module

Ensure the node entity is a ContentEntityInterface before proceeding. This prevents potential errors when invalid or unexpected entity types are encountered in the route parameters.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Refactor translation attachment logic for API service cards

Removed redundant checks and moved the 'api_service_card' logic to a more appropriate location. This ensures cleaner code and avoids unnecessary processing for this specific content type.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Update API service card entity form display configuration

Added support for the "allowed_formats" module and updated third-party settings to include formatting guidelines and maxlength configurations. Adjusted field settings and introduced a new "translation" field to the entity form display. These changes enhance content input flexibility and management.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Enable field translations for API Service Card. The rest fields changes to false automatically because we don't use Core translations for them.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Refactor language configuration for improved negotiation.

Added "language_content" to configurable types and updated method weights for content negotiation. Changed session language parameter from "language" to "content_lang" for better clarity and alignment with content language handling.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
Add drupal/coffee module to the project

Included the Drupal Coffee module (v2.0.1) in the dependencies and updated `core.extension.yml` to register it. This module provides an Alfred-like search box to enhance navigation within the site.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
Restore core.extension.yml.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Add missing config file.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

arthurbaghdas commented:
<!--
  1 failure:  Add a changelog Y...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Add a changelog YAML file to this PR</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9b693987907f0ad4d4d9f3788230fd7df818975d
</p>

dstorozhuk committed:
DP-39093: Remove Coffee module configuration and dependencies

The Coffee module has been fully removed from the configuration files, including its settings and references in `core.extension.yml`. This cleanup ensures unused modules do not remain in the codebase, reducing clutter and potential maintenance overhead.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
Revert DP-39093: Enable field translations for API Service Card. The rest fields changes to false automatically because we don't use Core translations for them.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Make API Service Card fields translatable

Updated multiple fields in the API Service Card configuration to support translations by setting them as translatable. Adjusted the default text format for "IDV Return URLs" from plain_text to basic_html to allow more flexible content styling. These changes enhance localization and improve content management capabilities.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Adjust language content negotiation priorities

Added a priority value for 'language-content-entity' in the language negotiation settings. This ensures more precise control over the language determination process, improving content handling in multilingual environments.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Update language negotiation settings in config

Removed the `language-content-entity` method from enabled negotiation methods. This simplifies the configuration and ensures the language methods align with current requirements.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
Add PHPStorm metadata for modules, entity types, and date formats

Introduced `.phpstorm.meta.php` files to provide improved autocompletion and argument suggestions in PHPStorm. These include metadata for registered modules, entity types, storage methods, date formats, and themes. This enhances the developer experience when working with Drupal code in the IDE.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Update media translations route path in routing file

Changed the route path for media translations from '/media/{media}/translations' to '/media/{media}/mass-translations'. This ensures better consistency and clarity in route definitions. Updated routing file aligns with controller functionality.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Add custom content translation controller and route subscriber

Introduce a custom `ContentTranslationController` to modify translation overview functionality, including adding language parameters to links. Implement a route subscriber to override the default controller for `entity.node.content_translation_overview` when session-based language negotiation is enabled.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
DP-39093: Add installation script to enable content_translation module

Introduced a .install file for the mass_translations module to manage installation and update tasks. The initial update hook automatically enables the content_translation module, ensuring necessary dependencies are active.

Signed-off-by: Dmytro Storozhuk <dima@itech4web.com>

dstorozhuk committed:
Merge branch 'develop' into feature/DP-39093-Service-Card-API-Fix-default-Drupal-Translations-for-Service-Card-API-content-type

dstorozhuk commented:
close in favor of https://github.com/massgov/openmass/pull/2976
